var/spack/repos/builtin/packages/cbtf-krell/package.py

```diff
@@ -262,3 +262,10 @@ def setup_environment(self, spack_env, run_env):
 
         run_env.set('CBTF_MRNET_BACKEND_PATH',
                     self.prefix.sbin.cbtf_libcbtf_mrnet_backend)
+
+        run_env.prepend_path('PATH', self.spec['libmonitor'].prefix.bin)
+        run_env.prepend_path('PATH', self.spec['papi'].prefix.bin)
+        run_env.prepend_path('PATH', self.spec['mrnet'].prefix.bin)
```

Comment:
Maybe these dependencies should be added as `type='run'`, then autoloaded?

Reply:
Let me experiment with that.   My work colleague asked me to add these because we have utilities in the bin directory for the above packages that we need access to.

Reply:
Making papi type run:   depends_on("papi@5.4.1:", type=('build', 'link', 'run'))
Didn't seem to work.   Is there something else I need to change to make the bin directory of papi available when loading the cbtf-krell module?   Thanks!

Reply:
See https://spack.readthedocs.io/en/latest/module_file_support.html#autoload-dependencies. I opened a bug for why this wasn't automatic for all run dependencies in #8639.

Reply:
@adamjstewart I think for us, option 1) in #8639 would be ideal.   But, for now it would be better for us to hard code the paths into the module files instead of having to depend on the user enter the auto load commands into a configuration file.   Can we do that until #8639 is merged?